### DISCUSSION TITLE
what should I do with this?

### DISCUSSION THREAD

**Discussion by MrLoney-dev** [maurosoria/dirsearch#1179] [General]
what should I do with this?

Traceback (most recent call last):
  File "dirsearch.py", line 23, in <module>
    from pkg_resources import DistributionNotFound, VersionConflict
ImportError: No module named pkg_resources

**Comment by shelld3v** [comment]
Can you try `pip install --upgrade setuptools`?

**Reply by MrLoney-dev** [reply]
what this succes?


Installing collected packages: setuptools
Successfully installed setuptools-65.2.0

**Reply by shelld3v** [reply]
Do you still face the same problem now?

**Reply by kimoji666** [reply]
> 您现在仍然面临同样的问题吗？

同样的问题，运行pip install --upgrade setuptools 还是不行

**Comment by vtino17** [comment]
This error means \pkg_resources\ module is missing — it's part of \setuptools\ which isn't always installed by default in newer Python versions.

**Fix:**
\\\
pip install setuptools
\\\

Then try running dirsearch again.

If you're using a virtual environment, recreate it with system-site-packages:
\\\
python -m venv --system-site-packages venv
.\venv\Scripts\activate  # or source venv/bin/activate on Linux
pip install -r requirements.txt
\\\

**Note:** \pkg_resources\ was deprecated in Python 3.12+, so dirsearch may need an update to use \importlib.metadata\ instead. For now, installing setuptools resolves it.
